### PR TITLE
[1/X] Add inline documentation to ApolloAPI

### PR DESCRIPTION
**Pull request by AnthonyMDev**
Adds the first bit of inline documentation.

Had to change GraphQLEnum `__unknown` case to remove the `__` prefix because DocC won't pick it up otherwise right now. https://github.com/apple/swift-docc/issues/342

This is going to be an issue for other properties like `__typename` that we probably aren't going to be willing to change, so I'm looking for a workaround there. If I can't find one, we're just going to have to write a not about it in the Docs... :/

### CODE REVIEW

**Comment by calvincestari** [comment]
Where can I find a deploy preview for these changes, is there one? Neither of the Netlify deploy preview links above seem to work; they've both exiting early because of "no content change"?

File: Sources/ApolloAPI/GraphQLNullable.swift [hunk 1]
```diff
@@ -111,17 +185,48 @@ extension Dictionary: _InitializableByDictionaryLiteralElements {
...
+  /// // Instead of
+  /// value = .init(.case(.NEWHOPE))
+  /// ```
+  /// - Parameter caseValue: A case of a generated ``EnumType`` to initialize a
+  /// `GraphQLNullable<GraphQLEnum<T>` with.
   @inlinable init<T: EnumType>(_ caseValue: T) where Wrapped == GraphQLEnum<T> {
     self = .some(Wrapped(caseValue))
   }
 
+  /// Intializer for use with an ``InputObject`` value.
```
Comment (calvincestari) [inline Sources/ApolloAPI/GraphQLNullable.swift:204]:
```suggestion
  /// Intializer for use with an ``InputObject`` value.
  ///
```

File: Sources/ApolloAPI/GraphQLNullable.swift [hunk 2]
```diff
@@ -2,13 +2,61 @@ import Foundation
...
+/// ### Literals, Enums, and InputObjects
+/// A ``GraphQLNullable`` can be converted from most scalar literals, ``GraphQLEnum``, and any
+/// generated ``InputObject`` automatically.
+///
+/// ```swift
+/// public init(inputValue: GraphQLNullable<String>)
+/// .init(inputValue: "InputValueString")
+///
+/// public init(episode: GraphQLNullable<GraphQLEnum<StarWarsAPI.Episode>>)
+/// .init(episode: .NEWHOPE)
```
Comment (calvincestari) [inline Sources/ApolloAPI/GraphQLNullable.swift:22]:
```suggestion
/// .init(episode: .init(.NEWHOPE))
```

File: Sources/ApolloAPI/InputValue.swift [hunk 3]
```diff
@@ -17,17 +18,20 @@ public indirect enum InputValue {
   case variable(String)
 
   /// A GraphQL "`List`" input value.
-  /// - See: [GraphQLSpec - Input Values - List Value](http://spec.graphql.org/June2018/#sec-List-Value)
+  /// # See Also
+  /// [GraphQLSpec - Input Values - List Value](http://spec.graphql.org/June2018/#sec-List-Value)
```
Comment (calvincestari) [inline Sources/ApolloAPI/InputValue.swift:22]: I think we should link to the latest non-working draft release of the spec.

```suggestion
  /// [GraphQLSpec - Input Values - List Value](http://spec.graphql.org/October2021/#sec-List-Value)
```

File: Sources/ApolloAPI/InputValue.swift [hunk 4]
```diff
@@ -17,17 +18,20 @@ public indirect enum InputValue {
...
   case list([InputValue])
 
   /// A GraphQL "`InputObject`" input value. Represented as a dictionary of input values.
-  /// - See: [GraphQLSpec - Input Values - Input Object Values](http://spec.graphql.org/June2018/#sec-Input-Object-Values)
+  /// # See Also
+  /// [GraphQLSpec - Input Values - Input Object Values](http://spec.graphql.org/June2018/#sec-Input-Object-Values)
```
Comment (calvincestari) [inline Sources/ApolloAPI/InputValue.swift:27]:
```suggestion
  /// [GraphQLSpec - Input Values - Input Object Values](http://spec.graphql.org/October2021/#sec-Input-Object-Values)
```

File: Sources/ApolloAPI/InputValue.swift [hunk 5]
```diff
@@ -17,17 +18,20 @@ public indirect enum InputValue {
...
   case object([String: InputValue])
 
   /// A null input value.
   ///
   /// A null input value indicates an intentional inclusion of a value for a field argument as null.
-  /// - See: [GraphQLSpec - Input Values - Null Value](http://spec.graphql.org/June2018/#sec-Null-Value)
+  /// # See Also
+  /// [GraphQLSpec - Input Values - Null Value](http://spec.graphql.org/June2018/#sec-Null-Value)
```
Comment (calvincestari) [inline Sources/ApolloAPI/InputValue.swift:34]:
```suggestion
  /// [GraphQLSpec - Input Values - Null Value](http://spec.graphql.org/October2021/#sec-Null-Value)
```

File: Sources/ApolloAPI/JSONDecodingError.swift [hunk 6]
```diff
@@ -1,9 +1,20 @@
 import Foundation
 
+/// An error thrown while decoding `JSON`.
+///
+/// This error should be thrown when a ``JSONDecodable/init(jsonValue:)`` fails.
+/// `GraphQLExecutor` and `ApolloStore` may also throw this error when decoding a `JSON` fails.
 public enum JSONDecodingError: Error, LocalizedError, Hashable {
+  /// A value that is expected to be present is missing from the ``JSONObject``.
   case missingValue
+  /// A value that is non-null has a `null`value.
```
Comment (calvincestari) [inline Sources/ApolloAPI/JSONDecodingError.swift:10]:
```suggestion
  /// A value that is non-null has a `null` value.
```

File: Sources/ApolloAPI/ScalarTypes.swift [hunk 7]
```diff
@@ -1,18 +1,20 @@
 /// An abstract protocol that a GraphQL "`scalar`" type must conform to.
 ///
-/// - See: [GraphQL Spec - Scalars](http://spec.graphql.org/June2018/#sec-Scalars)
+/// # See Also
+/// [GraphQL Spec - Scalars](http://spec.graphql.org/June2018/#sec-Scalars)
```
Comment (calvincestari) [inline Sources/ApolloAPI/ScalarTypes.swift:4]:
```suggestion
/// [GraphQL Spec - Scalars](http://spec.graphql.org/October2021/#sec-Scalars)
```

File: Sources/ApolloAPI/ScalarTypes.swift [hunk 8]
```diff
@@ -1,18 +1,20 @@
...
-///
 /// Conforming types are:
 ///   * `String`
 ///   * `Int`
 ///   * `Bool`
 ///   * `Float`
 ///   * `Double`
+///
+/// # See Also
+/// [GraphQL Spec - Scalars](http://spec.graphql.org/June2018/#sec-Scalars)
```
Comment (calvincestari) [inline Sources/ApolloAPI/ScalarTypes.swift:17]:
```suggestion
/// [GraphQL Spec - Scalars](http://spec.graphql.org/October2021/#sec-Scalars)
```

File: Sources/ApolloAPI/ScalarTypes.swift [hunk 9]
```diff
@@ -31,7 +33,8 @@ extension Double: ScalarType {}
 /// implementation of a custom scalar in the generated file. *Changes to generated custom scalar
 /// types will not be overwritten when running code generation again.*
 ///
-/// - See: [GraphQL Spec - Scalars](http://spec.graphql.org/June2018/#sec-Scalars)
+/// # See Also
+/// [GraphQL Spec - Scalars](http://spec.graphql.org/June2018/#sec-Scalars)
```
Comment (calvincestari) [inline Sources/ApolloAPI/ScalarTypes.swift:37]:
```suggestion
/// [GraphQL Spec - Scalars](http://spec.graphql.org/October2021/#sec-Scalars)
```

File: Sources/ApolloAPI/SchemaTypes/Interface.swift [hunk 10]
```diff
@@ -1 +1,3 @@
-open class Interface: CacheEntity, ParentTypeConvertible { }
+/// An abstract base class inherited by interface in a generated GraphQL schema.
```
Comment (calvincestari) [inline Sources/ApolloAPI/SchemaTypes/Interface.swift:1]:
```suggestion
/// An abstract base class inherited by interfaces in a generated GraphQL schema.
```

File: Sources/ApolloAPI/SelectionSet.swift [hunk 11]
```diff
@@ -10,8 +10,12 @@ public protocol AnySelectionSet: SelectionSetEntityValue {
   /// This may be a concrete type (`Object`) or an abstract type (`Interface`, or `Union`).
   static var __parentType: ParentType { get }
 
+  /// The data of the underlying GraphQL object represented by generated selection set.
   var __data: DataDict { get }
 
+  /// Designated Initializer
```
Comment (calvincestari) [inline Sources/ApolloAPI/SelectionSet.swift:16]:
```suggestion
  /// Designated Initializer
  ///
```

**Review by calvincestari — APPROVED** [review]
The documentation content looks good and I'm happy for this PR to be merged. I couldn't find a link to the generated docs though so I can't comment on any rendered layout.

File: Sources/ApolloAPI/AnyHashableConvertible.swift [hunk 12]
```diff
@@ -1,11 +1,21 @@
 import Foundation
-import CloudKit
```
[the remark this answers is not in the document]
Comment (AnthonyMDev) [reply Sources/ApolloAPI/AnyHashableConvertible.swift:2]: I have no clue. lol will fix

**Pull request merged**